src/platform/backends/hyperv_api/hcs_virtual_machine_factory.cpp

```diff
@@ -0,0 +1,291 @@
+        network_params.name = vswitch_name;
+        network_params.type = hcn::HcnNetworkType::Transparent();
+        network_params.guid = utils::make_uuid(network_params.name).toStdString();
+        hcn::HcnNetworkPolicy policy{hcn::HcnNetworkPolicyType::NetAdapterName(),
+                                     hcn::HcnNetworkPolicyNetAdapterName{intf.id}};
+        network_params.policies.push_back(policy);
+        return network_params;
+    }();
+
+    const auto& [status, status_msg] = HCN().create_network(params);
+
+    if (status || static_cast<HRESULT>(status) == HCN_E_NETWORK_ALREADY_EXISTS)
+    {
+        return params.name;
+    }
+
+    return {};
+}
+
+VirtualMachine::UPtr HCSVirtualMachineFactory::clone_vm_impl(const std::string& source_vm_name,
+                                                             const multipass::VMSpecs& src_vm_specs,
+                                                             const VirtualMachineDescription& desc,
+                                                             VMStatusMonitor& monitor,
+                                                             const SSHKeyProvider& key_provider)
+{
+
+    const fs::path src_vm_instance_dir{get_instance_directory(source_vm_name).toStdWString()};
+
+    if (!fs::exists(src_vm_instance_dir))
+    {
+        throw std::runtime_error{"Source VM instance directory is missing!"};
+    }
+
+    std::optional<fs::path> src_vm_vhdx{std::nullopt};
+
+    for (const auto& entry : fs::directory_iterator(src_vm_instance_dir))
+    {
+        const auto& extension = entry.path().extension();
+        if (extension == ".vhdx")
+        {
+            src_vm_vhdx = entry.path();
+            break;
+        }
+    }
+
+    if (!src_vm_vhdx.has_value())
+    {
+        throw std::runtime_error{"Could not locate source VM's vhdx file!"};
+    }
+
+    // Copy the VHDX file.
+    virtdisk::CreateVirtualDiskParameters clone_vhdx_params{};
+    clone_vhdx_params.predecessor = virtdisk::SourcePathParameters{src_vm_vhdx.value()};
+    clone_vhdx_params.path = desc.image.image_path.toStdString();
+    clone_vhdx_params.size_in_bytes = 0; // use source disk size
+
+    const auto& [status, msg] = VirtDisk().create_virtual_disk(clone_vhdx_params);
+
+    if (!status)
+    {
+        throw std::runtime_error{"VHDX clone failed."};
+    }
+
+    return create_virtual_machine(desc, key_provider, monitor);
+}
+
+std::vector<NetworkInterfaceInfo> HCSVirtualMachineFactory::get_adapters()
+{
+    std::vector<NetworkInterfaceInfo> ret;
+    for (auto& item : MP_PLATFORM.get_network_interfaces_info())
+    {
+        auto& net = item.second;
+        if (const auto& type = net.type; type == "Ethernet" || type == "IEEE 802.11")
+        {
+            net.needs_authorization = true;
+            ret.emplace_back(std::move(net));
+        }
+    }
+
+    return ret;
+}
+
+std::vector<NetworkInterfaceInfo> HCSVirtualMachineFactory::networks() const
+{
+    return get_adapters();
```

Comment:
This implementation is different from the old Hyper-V and QEMU. There could be an argument for not exposing bridges/switches in Multipass at all, and just attach to the end card, although that would prevent custom switches/bridges and void things like [our recipe for static IPs](https://documentation.ubuntu.com/multipass/latest/how-to-guides/manage-instances/configure-static-ips/#configure-static-ips)).

In short, we could even discuss a different contract, but should implement it consistently don't you think?